src/app/app.component.html

```diff
@@ -3,16 +3,8 @@
   <p slot="right-content"></p>
 </mo-header>
 <div *ngIf="weatherData.length==locLength">
-  <app-weather-data  *ngFor="let a of weatherData; index as i" [data]="weatherData[i]" ></app-weather-data>
+  <app-button *ngFor="let a of weatherData; index as i" [i]="i" [d]="weatherData[i].location.name" (displayData)="displayData($event, i)" ></app-button>
+  <div *ngIf="local!=null">
+      <app-weather-data   [data]="weatherData[local]" ></app-weather-data>
+  </div>
 </div>
```

Comment:
It'd actually be possible to do this without an additional component, using some of angular's features:

in the app.component, a variable called something like 'locationID'

`<div *ngIf="weatherData.length==locLength">
   <ng-container *ngFor="let a of weatherData; index as i">
      <button (click)="locationID = i`>weatherData[i].location.name</button
   </ng-container>
   <app-weather-data *ngIf="locationID " [data]="weatherData[locationID]"></app-weather-data>
 </div>